### ISSUE TITLE
Full Text Search Integration

### ISSUE THREAD

**Issue by tzach** [scylladb/scylla#1083]
Cassandra integrate with tools like Lucene / Solr / ES 
Similar integration with Scylla will be useful for many users.

**Comment by ddorian** [comment]
Do you have any plans on tackling this ?

Meaning implementing another search-lib (is there any that work with async writes on xfs?)
or implement something external like datastax-solr
or implement something different external like https://github.com/vroyer/elassandra
or do your own inverted-index-lib 
or something else?

**Comment by gkze** [comment]
This would be interesting to see

**Comment by colundrum** [comment]
Why not the possibility to use [Sphinx](http://sphinxsearch.com/about/sphinx/) ?
It's written in C++.

**Comment by ddorian** [comment]
New inverted index libraries have come up:

https://github.com/phaistos-networks/Trinity
https://github.com/tantivy-search/tantivy

Scylladb can integrate with one of them and do something like `elassandra` . 

Still, the best way would be what `couchbase` does, which is separate nodes by their `type` (for most efficient hardware uses). So you would have 1 cluster where some nodes are in-memory-pedis (ram, ok-disk, ok-cpu), some are search-engine (ram+cpu), some are columnar-store (hdd+cpu), some are scylladb (ssd+ram+cpu). Some type of queue between them and some type of spark sparkle on top. (don't know if it makes sense + needs alot dev-time)

**Comment by fulmicoton** [comment]
I'm available if you have questions about tantivy.

**Comment by didip** [comment]
Sphinx on Scylladb would be glorious.

**Comment by Remember2015** [comment]
> I'm available if you have questions about tantivy.

So is there any solution or possible way  to integrate tantivy / sphix with scylla?

**Comment by dorlaor** [comment]
We haven't look at it yet, you're welcome to give it a try

**Comment by elgalu** [comment]
It seems for now one needs to ingest the data into both Scylla and Elasticsearch using dual writes and then perform the textual search:
https://www.scylladb.com/2019/03/07/scylla-and-elasticsearch-part-two-practical-examples-to-support-full-text-search-workloads/

**Comment by dorlaor** [comment]
Right, Scylla doesn't have full text search, we haven't gotten to it..
We do have a LIKE operator which runs a scan on several partitions
and can retrieve a regular expression match. It's not an index but can be
a poor man's replacement to search in a relatively small range/partition.

**Comment by dorlaor** [comment]
I think that you can treat scan/range as a regular operation * N where N is
the
number of partitions. It's not accurate, since scan is efficient in terms
of server
side only processing but the partitions are scattered in the cluster.
LIKE as an operation, isn't an index but more of a regex. If you can live
with it,
use it. Give it a try if you have it running. It's a poor's man text search
